app/web_ui/src/lib/stores/rag_progress_store.ts

```diff
@@ -25,45 +26,103 @@ interface RagConfigurationProgressState {
   last_started_rag_config_id: string | null
 }
 
+interface ProjectRagProgressState {
+  projects: Record<string, RagConfigurationProgressState>
+}
+
 export const ragProgressStore = createRagProgressStore()
 
-export const allRagConfigs = derived(ragProgressStore, ($store) => {
-  return sortRagConfigs(Object.values($store.rag_configs), "created_at")
-})
+// derived store that provides the rag state for the currently selected project
+export const currentProjectRagProgressStore = derived(
+  [ragProgressStore, ui_state],
+  ([ragProgress, uiState]) => {
+    const currentProjectId = (uiState as UIState).current_project_id
```

Comment:
We should remove this. The library has a project ID in the URL, which may or may not be the current project. We should respect the exact project they link to, and ignore current_project_id.
In general: current_project_id is deprecated. There's going to be one layout allowed to use it (just on fresh page load, just to redirect to a proper URL with project_id in it), but the URL based parameters are the source of truth, and current_project_id is the source of bugs. It gets messed up in lots of ways (deeplinks, multiple tabs, etc)
Just use something like $: progressStore = ragProgressStore[$page.params.project_id] in callers, it works great